tests/unit_tests/data_steward/resources_test.py

```diff
@@ -65,3 +68,76 @@ def test_cdm_schemas(self):
             result_vocab_tables,
             msg='Vocabulary tables %s should not be in result of cdm_schemas()'
             % result_vocab_tables)
+
+    def test_fields_for(self):
+        """
+        Testing that fields for works as expected with sub-directory structures.
+        """
+        # preconditions
+
+        # test
+        actual_fields = resources.fields_for('person')
+
+        # post conditions
+        person_path = os.path.join(resources.base_path, 'resource_files',
+                                   'fields', 'person.json')
+        with open(person_path, 'r') as fp:
+            expected_fields = json.load(fp)
+
+        self.assertEqual(actual_fields, expected_fields)
+
+    @mock.patch('resources.os.walk')
+    def test_fields_for_duplicate_files(self, mock_walk):
```

Comment:
I'm going to attempt to explain this test function so I know what's going on --

The walk results are:
`foo/bar/duplicate.json`
`foo/bar/unique1.json`
`foo/bar/baz/duplicate.json`
`foo/bar/baz/unique.json`

For the first test this isn't good because we have two json files `duplicate.json` which would throw a RuntimeError.

For the second test (I'm less sure about), it's ensuring that `baz/duplicate.json` returns the json with the right fields inside, right?

Reply:
good questions.  i'll add some comments to this test for future reference.  I'll address your questions here and we can talk via slack if you want.

Python's `os.walk` [returns a list of tuples](https://docs.python.org/3/library/os.html#os.walk).  Each tuple is of the form `(directory_string, list of contained directories, list of files)`.  So your directory structure is correct.  

The `fields_for` implementation doesn't care about the contained directories and pointedly ignores them using `_`.  Because they are ignored, that is represented as a blank list in `walk_results`.  If this were an accurate depiction of `os.walk` based on the given structure, then walk results should be:
```suggestion
walk_results = [
    (os.path.join('foo', 'bar'), ['baz'], ['duplicate.json', 'unique1.json']),
    (os.path.join('foo', 'bar', 'baz'), [], ['duplicate.json', 'unique2.json'])
]
```
I'll probably make this change also.

The first test highlights a concern @mark-velez had.  If we have two files with the same name in different sub-directories and don't indicate which one we expect, then `fields_for` doesn't know which one is correct.  It could make an assumption and return either the first or last it finds.  But instead of building in assumptions, Mark's suggestion of throwing an error is better.  So, with this `walk_results` and the existing function signature, `duplicate.json` would be found multiple times.  This test verifies the error is raised (which is what we want), so the test passes only when the error is raised.

The second test uses the `sub_path` parameter to dictate which sub-directory to walk.  `fields_for` will create the path `<path_to_curation_repo/curation/data_steward/resource_files/fields/<sub_path_parameter>`.  This directory will be walked.  When this directory is walked, `duplicate.json` is found and the upper level duplicate file is not.  So, the file is unique and a `RuntimeError` isn't raised.  The file is then opened and the json response object is returned.  The patch provides mock data for the file open function to read and return.  This just lets the test wrap up nicely by checking the equality of the expected json object with what it actually returns.

Reply:
Yeah adding some comments would be helpful here :D also, that change makes sense to me.